tests/cli/functional/conftest.py

```diff
@@ -16,20 +16,107 @@
+from tests.constants import (
+    KEYFILE_NAME_TEMPLATE,
+    MOCK_KEYSTORE_PATH,
+    MOCK_PROVIDER_URI,
+    NUMBER_OF_MOCK_KEYSTORE_ACCOUNTS
+)
 from tests.fixtures import _make_testerchain, make_token_economics
-from tests.mock.agents import FAKE_RECEIPT, MockContractAgency
+from tests.mock.agents import MockContractAgency, MockContractAgent
 from tests.mock.interfaces import MockBlockchain, make_mock_registry_source_manager
+from tests.utils.config import (
+    make_alice_test_configuration,
+    make_bob_test_configuration,
+    make_ursula_test_configuration
+)
+from tests.utils.ursula import MOCK_URSULA_STARTING_PORT
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_contract_agency(monkeypatch, module_mocker, token_economics):
+    monkeypatch.setattr(ContractAgency, 'get_agent', MockContractAgency.get_agent)
+    module_mocker.patch.object(EconomicsFactory, 'get_economics', return_value=token_economics)
+    mock_agency = MockContractAgency()
+    yield mock_agency
+    mock_agency.reset()
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_token_agent(mock_testerchain, token_economics, mock_contract_agency):
+    return mock_contract_agency.get_agent(NucypherTokenAgent)
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_staking_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(StakingEscrowAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_adjudicator_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(AdjudicatorAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_policy_manager_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(PolicyManagerAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_multisig_agent(mock_testerchain, token_economics, mock_contract_agency):
+    mock_agent = mock_contract_agency.get_agent(MultiSigAgent)
+    return mock_agent
+
+
+@pytest.fixture(scope='function', autouse=True)
+def mock_worklock_agent(mock_testerchain, token_economics, mock_contract_agency):
+    economics = token_economics
+    mock_agent = mock_contract_agency.get_agent(WorkLockAgent)
+    # Customize the mock agent
+    spec = {'boosting_refund': economics.worklock_boosting_refund_rate,
+            'slowing_refund': 1,  # TODO: another way to get this value?
+            'start_bidding_date': economics.bidding_start_date,
+            'end_bidding_date': economics.bidding_end_date,
+            'end_cancellation_date': economics.cancellation_end_date,
+            'minimum_allowed_bid': economics.worklock_min_allowed_bid,
+            'lot_value': economics.worklock_supply}
+    for method_name, return_value in spec.items():
+        setattr(mock_agent.__class__, method_name, return_value)
```

Comment:
can we remove strings and use `mock_agent.boosting_refund.return_value = economics.worklock_boosting_refund_rate` or equivalent from `PropertyMock`?

Reply:
Yes.  Great feedback here.  We want to eliminate the bulk of mock strings for better maintainability.

Reply:
:heavy_check_mark: - Let me know if this is enough.